src/encoded/static/components/cart/manager.js

```diff
@@ -293,17 +288,24 @@ class NameCartButtonComponent extends React.Component {
      * function to execute the operation depends on whether we're creating or renaming a cart.
      */
     handleSubmit() {
-        const { create, onCreate, onRename, updateCartManager } = this.props;
+        const { create, onCreate, onRename, onSwitchCart, updateCartManager, user } = this.props;
         const { newName, newIdentifier } = this.state;
         const submitFunc = create ? onCreate : onRename;
-        submitFunc(newName.trim(), newIdentifier).then(() => {
-            // Successfully changed the cart name/identifier. Indicate the cart manager needs
-            // reloading so it displays the new cart information.
+        submitFunc(newName.trim(), newIdentifier).then((cartAtId) => {
```

Comment:
You are modifying a parameter. That is an anti-pattern.

Reply:
You mean trim()? According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim#Description), trim() makes a new string.